extensions/register.md

```diff
@@ -0,0 +1,127 @@
+---
+title: The REGISTER command
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source of frustration for many participants
+for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal, this effort aims to be more
+appealing to implementors by being severely restricted in scope. The only register-verify flows supported
+are those already in use in the wild. The authors hope that this stripped-down solution will serve as the
+basis for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence signifies that the server
+accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value pairs; the format is intended
+to follow the precedent set by [the multiline draft][multiline].
+
+The `draft/register` capability SHOULD NOT be requested. Servers MAY `NAK` any such request. The
+`REGISTER` command MUST be accepted regardless of any attempt to negotiate or disable the capability.
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early registration, so it
+   MUST NOT use `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address to process
+
+### Commands
+
+    REGISTER {<email> | "*"} <password>
```

Comment:
Thoughts on

```suggestion
The defined keys are:

 * `before-connect` - if present, indicates the server supports early registration, so it
   MUST NOT use `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
 * `verify` - if present, registrations require verification to proceed
 * `verify-email` - if present, verification can be done via email
 * `verify-tel` - if present, verification can be done via telephone

### Commands

    REGISTER {verify-mean=<verify-value> | "*"} <password>
```

?

It'd allow something like

```
S: CAP * LS :register=before-connect,verify,verify-email,verify-tel
C: REGISTER tel=+33612345678 :a very secure password
```

Vendor verification means could be added like so:

```
S: CAP * LS :register=before-connect,verify,verify-email,vendor/verify-matrix
C: REGISTER vendor/matrix=@freenode_pinkieval:matrix.org :though this should be strictly checked
```

Reply:
I don't think we should try to make this spec as generic as ACC was.

Reply:
I agree with @emersion. I'll add a requirement that clients should ignore unknown keys, so future specification can extend this one by adding keys.